### PR TITLE
Split htaccess / Add AdvServer to J

### PR DESCRIPTION
**Pull request by conconnl**
As suggested by Michael in  https://github.com/joomla/joomla-websites/pull/1443
Renamed the first baseline to General as it can be used on none-joomla website and created a new one specifically for Joomla

### CODE REVIEW

File: configuration-baselines/.htaccess-joomla [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+###########################################
+
+###########################################
+# ========== Add redirects here ===========
+
+###########################################
+
+###########################################
+# === Optimal default expiration time ===
+<IfModule mod_expires.c>
```
Comment (mbabker) [inline configuration-baselines/.htaccess-joomla:48]: The `mod_expires`, `mod_deflate`, and `mod_gzip` stuff can be in the general config.  These sections are dealing with defining "optimal" (opinionated) cache timeframes for different asset types and optimizing resource delivery.

The stuff about not caching `administrator/index.php` or rewrite rules for the CMS filesystem structure or query strings for the CMS' non-SEF configuration is what rightfully belongs in this file.

File: configuration-baselines/.htaccess-joomla [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+###########################################
+# ===== Advanced server protection  =======
+## Disable PHP Easter Eggs
+RewriteCond %{QUERY_STRING} \=PHP[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12} [NC]
+RewriteRule .* - [F]
+
+#### Front-end protection
+## Allow limited access for certain directories with client-accessible content
+RewriteRule ^(components|modules|templates|images|plugins|media|libraries|media/jui/fonts|cache/com_comment/init)/.*\.(jpe|jpg|jpeg|jp2|jpe2|png|gif|bmp|css|js|swf|html|mpg|mp3|mpeg|mp4|avi|wav|ogg|ogv|xls|xlsx|doc|docx|ppt|pptx|zip|rar|pdf|xps|txt|7z|svg|odt|ods|odp|flv|mov|ico|htm|ttf|woff|eot|JPG|JPEG|PNG|GIF|CSS|JS|TTF|WOFF|EOT)$ - [L]
```
Comment (mbabker) [inline configuration-baselines/.htaccess-joomla:158]: 1. What extensions are creating publicly accessible resources in the `cache` directory?  AFAIK there is no "com_comment" on any .org site, and whitelisting paths in the `cache` directory raises a red flag about extension code.

2. SVG should be allowed here.

3. You have some file extensions duplicated as all caps and lowercased.  It'd probably be good if the extension list were case insensitive so you're not duplicating the list.

**Comment by conconnl** [comment]
I'm now totally confused.
First you tell me to make two files, one for Joomla and one for other sites, now you want me to remove the things as well for Joomla. So people manually need to merge the files.

The lines about Frontend and many others generated by a professional tool, from one of our valued community members. If those are not good, it even means we need to start removing the lines generated by the tool on the different properties maintained by the different webmasters.

**Comment by mbabker** [comment]
Sorry, I didn't realize that the Joomla file was basically "everything in general plus Joomla specific addons".  I was thinking that the Joomla file was only additions to the general list, hence that's how I responded.

I know where the frontend list is generated from.  That doesn't mean it can't be improved upon.

**Comment by mbabker** [comment]
> Removing a few words case-sensitive words is easy, does this have any impact?

My regex is terrible, but it should be possible to specify that a list is matched in a case insensitive manner (if not the entire regex string).  It avoids the bit where you have what looks to be 25 or 30 file extensions in lowercase and 10 or so in ALL CAPS.

> Removing com_comment is understandable, but does it mean the whole part "|cache/com_comment/init" or just com_comment ?

Removing `|cache/com_comment/init`.

File: configuration-baselines/.htaccess-joomla (hunk 2 above)
Comment (conconnl) [reply configuration-baselines/.htaccess-joomla]: I don't know Regex the rest is done.
I found the following on Google
image

Comment (conconnl) [reply configuration-baselines/.htaccess-joomla]: As I understand we then should change:
`RewriteRule ^(components|modules|templates|images|plugins|media|libraries|media/jui/fonts)/.*\.(jpe|jpg|jpeg|jp2|jpe2|svg|png|gif|bmp|css|js|swf|html|mpg|mp3|mpeg|mp4|avi|wav|ogg|ogv|xls|xlsx|doc|docx|ppt|pptx|zip|rar|pdf|xps|txt|7z|svg|odt|ods|odp|flv|mov|ico|htm|ttf|woff|eot|JPG|JPEG|PNG|GIF|CSS|JS|TTF|WOFF|EOT)$ - [L]`

Too
`RewriteRule ^(components|modules|templates|images|plugins|media|libraries|media/jui/fonts)/.*\.(jpe|jpg|jpeg|jp2|jpe2|svg|png|gif|bmp|css|js|swf|html|mpg|mp3|mpeg|mp4|avi|wav|ogg|ogv|xls|xlsx|doc|docx|ppt|pptx|zip|rar|pdf|xps|txt|7z|svg|odt|ods|odp|flv|mov|ico|htm|ttf|woff|eot)$ - [L,NC]`

Comment (conconnl) [reply configuration-baselines/.htaccess-joomla]: @mbabker do you have any opinion about my last message?

File: configuration-baselines/.htaccess-joomla [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+## Disable PHP Easter Eggs
+RewriteCond %{QUERY_STRING} \=PHP[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12} [NC]
+RewriteRule .* - [F]
+
+#### Front-end protection
+## Allow limited access for certain directories with client-accessible content
+RewriteRule ^(components|modules|templates|images|plugins|media|libraries|media/jui/fonts)/.*\.(jpe|jpg|jpeg|jp2|jpe2|svg|png|gif|bmp|css|js|swf|html|mpg|mp3|mpeg|mp4|avi|wav|ogg|ogv|xls|xlsx|doc|docx|ppt|pptx|zip|rar|pdf|xps|txt|7z|svg|odt|ods|odp|flv|mov|ico|htm|ttf|woff|eot|JPG|JPEG|PNG|GIF|CSS|JS|TTF|WOFF|EOT)$ - [L]
+RewriteRule ^(components|modules|templates|images|plugins|media|libraries|media/jui/fonts)/ - [F]
+## Disallow front-end access for certain Joomla! system directories (unless access to their files is allowed above)
+RewriteRule ^includes/js/ - [L]
```
Comment (zero-24) [inline configuration-baselines/.htaccess-joomla:161]: ``includes/js/`` ? There is no ``includes/js/`` folder in the CMS?

File: configuration-baselines/.htaccess-joomla [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+RewriteRule ^(cache|includes|language|logs|log|tmp)/ - [F]
+RewriteRule ^(configuration\.php|CONTRIBUTING\.md|htaccess\.txt|joomla\.xml|LICENSE\.txt|phpunit\.xml|README\.txt|web\.config\.txt) - [F]
```
Comment (zero-24) [inline configuration-baselines/.htaccess-joomla:163]: Not sure about phpunit and contributig files given that they are not shipped with the core.

**Comment by zero-24** [comment]
@mbabker can you please take a final look here whether the requested changes has been implemented now?

**Comment by zero-24** [comment]
Merging as requested

**Pull request merged**